packages/x-data-grid/src/material/index.tsx

```diff
@@ -169,120 +174,54 @@ const StyledPagination = styled(MUIPagination)(({ theme }) => ({
-  densityCompactIcon: GridViewHeadlineIcon,
-  densityStandardIcon: GridTableRowsIcon,
-  densityComfortableIcon: GridViewStreamIcon,
-  exportIcon: GridDownloadIcon,
-  moreActionsIcon: GridMoreVertIcon,
-  treeDataCollapseIcon: GridExpandMoreIcon,
-  treeDataExpandIcon: GridKeyboardArrowRight,
-  groupingCriteriaCollapseIcon: GridExpandMoreIcon,
-  groupingCriteriaExpandIcon: GridKeyboardArrowRight,
-  detailPanelExpandIcon: GridAddIcon,
-  detailPanelCollapseIcon: GridRemoveIcon,
-  rowReorderIcon: GridDragIcon,
-  quickFilterIcon: GridSearchIcon,
-  quickFilterClearIcon: GridClearIcon,
-  columnMenuHideIcon: GridVisibilityOffIcon,
-  columnMenuSortAscendingIcon: GridArrowUpwardIcon,
-  columnMenuSortDescendingIcon: GridArrowDownwardIcon,
-  columnMenuUnsortIcon: null,
-  columnMenuFilterIcon: GridFilterAltIcon,
-  columnMenuManageColumnsIcon: GridViewColumnIcon,
-  columnMenuClearIcon: GridClearIcon,
-  loadIcon: GridLoadIcon,
-  filterPanelAddIcon: GridAddIcon,
-  filterPanelRemoveAllIcon: GridDeleteForeverIcon,
-  columnReorderIcon: GridDragIcon,
-  menuItemCheckIcon: GridCheckIcon,
-};
+const BasePagination = forwardRef<any, P['basePagination']>(function BasePagination(props, ref) {
+  const { onRowsPerPageChange, material, disabled, ...rest } = props;
+  const computedProps = React.useMemo(() => {
+    if (!disabled) {
+      return undefined;
+    }
+    return {
+      backIconButtonProps: { disabled: true },
+      nextIconButtonProps: { disabled: true },
+    };
+  }, [disabled]);
 
-const baseSlots: GridBaseSlots = {
-  baseAutocomplete: BaseAutocomplete,
-  baseBadge: MUIBadge,
-  baseCheckbox: React.forwardRef(BaseCheckbox),
-  baseCircularProgress: MUICircularProgress,
-  baseDivider: MUIDivider,
-  baseInput: BaseInput,
-  baseLinearProgress: MUILinearProgress,
-  baseMenuList: BaseMenuList,
-  baseMenuItem: BaseMenuItem,
-  baseTextField: BaseTextField,
-  baseButton: MUIButton,
-  baseIconButton: MUIIconButton,
-  baseTooltip: MUITooltip,
-  basePagination: BasePagination,
-  basePopper: BasePopper,
-  baseSelect: BaseSelect,
-  baseSelectOption: BaseSelectOption,
-  baseSkeleton: MUISkeleton,
-  baseSwitch: MUISwitch,
-};
+  const apiRef = useGridApiContext();
+  const rootProps = useGridRootProps();
+  const { estimatedRowCount } = rootProps;
 
-const materialSlots: GridBaseSlots & GridIconSlotsComponent = {
-  ...baseSlots,
-  ...iconSlots,
-};
+  return (
+    <StyledPagination
+      component="div"
+      onRowsPerPageChange={useEventCallback(
+        (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+          onRowsPerPageChange?.(Number(event.target.value));
+        },
+      )}
+      labelRowsPerPage={apiRef.current.getLocaleText('paginationRowsPerPage')}
+      labelDisplayedRows={(params) =>
+        apiRef.current.getLocaleText('paginationDisplayedRows')({
+          ...params,
+          estimated: estimatedRowCount,
+        })
+      }
+      // @ts-ignore Issue with material
```

Comment:
What is the issue here?

Reply:
I don't understand. The typecheck works if `{ ...material }` is not there, but fails if it is present. The prop `getItemAriaLabel` could be overriden via `material`, but I don't understand how that makes the previous typing fail. Do you think it's worth investigating?

Reply:
It was due to the `Pagination`/`TablePagination` props mix-up, should be fixed now